packages/flutter/lib/src/painting/text_style.dart

```diff
@@ -1153,8 +1176,77 @@ class TextStyle with Diagnosticable {
       );
     }
 
+    assert(() {
+      if (a.inherit == b.inherit) {
+        return true;
+      }
+
+      final List<String> nullFields = <String>[
+        if (a.foreground == null && b.foreground == null && a.color == null && b.color == null) 'color',
+        if (a.background == null && b.background == null && a.backgroundColor == null && b.backgroundColor == null) 'backgroundColor',
+        if (a.fontSize == null && b.fontSize == null) 'fontSize',
+        if (a.letterSpacing == null && b.letterSpacing == null) 'letterSpacing',
+        if (a.wordSpacing == null && b.wordSpacing == null) 'wordSpacing',
+        if (a.height == null && b.height == null) 'height',
+        if (a.decorationColor == null && b.decorationColor == null) 'decorationColor',
+        if (a.decorationThickness == null && b.decorationThickness == null) 'decorationThickness',
+      ];
+      if (nullFields.isEmpty) {
+        return true;
+      }
+
+      throw FlutterError.fromParts(<DiagnosticsNode>[
+        ErrorSummary('Failed to interpolate TextStyles with different inherit values.'),
+        ErrorSpacer(),
+        ErrorDescription('The TextStyles being interpolated were:'),
+        a.toDiagnosticsNode(name: 'from', style: DiagnosticsTreeStyle.singleLine),
+        b.toDiagnosticsNode(name: 'to', style: DiagnosticsTreeStyle.singleLine),
+        ErrorDescription(
+          'The following fields are unspecified in both TextStyles:\n'
+          '${nullFields.map((String name) => '"$name"').join(', ')}.\n'
+          'When "inherit" changes during the transition, these fields may '
+          'observe abrupt value changes as a result, causing "jump"s in the '
+          'transition.'
+        ),
+        ErrorSpacer(),
+        ErrorHint(
+          'In general, TextStyle.lerp only works well when both TextStyles have '
+          'the same "inherit" value, and set roughly the same fields.',
```

Comment:
```suggestion
          'the same "inherit" value, and specify roughly the same fields.',
```

Also: Do we really need to say "roughly" the same fields?

Reply:
removed roughly